test/gateway/eip20_gateway/deactivate_gateway.js

```diff
@@ -0,0 +1,54 @@
+const Gateway = artifacts.require("./EIP20Gateway.sol")
+    , BN = require('bn.js');
+
+const Utils = require('../../../test/test_lib/utils');
+
+
+contract('EIP20Gateway.deactivateGateway()', function (accounts) {
+
+    let gateway;
+    let organisation = accounts[2];
+
+    beforeEach(async function () {
+
+        let mockToken = accounts[0],
+            baseToken = accounts[1],
+            coreAddress = accounts[2],
+            bountyAmount = new BN(100);
+
+        gateway = await Gateway.new(
+            mockToken,
+            baseToken,
+            coreAddress,
+            bountyAmount,
+            organisation
+        );
+
+        let coGateway = accounts[5];
+        await  gateway.activateGateway(coGateway, {from: organisation});
+
+    });
+
+    it('should deactivate if activated', async function () {
+
+        assert((await gateway.deactivateGateway.call({from: organisation})));
```

Comment:
we can 
- split this to be more readable.
- use assert.strictEqual.
- add assert fail reason.

Reply:
These changes were introduced when i pulled the below branch to sync up :-
https://github.com/OpenSTFoundation/mosaic-contracts/pull/518/files